src/core/mesh/qgsmeshrenderersettings.cpp

```diff
@@ -424,6 +450,19 @@ void QgsMeshRendererSettings::readXml( const QDomElement &elem )
 
   QDomElement elemTriangularMesh = elem.firstChildElement( QStringLiteral( "mesh-settings-triangular" ) );
   mRendererTriangularMeshSettings.readXml( elemTriangularMesh );
+
+  QDomElement elemAveraging = elem.firstChildElement( QStringLiteral( "averaging-3d" ) );
+  if ( !elemAveraging.isNull() )
+  {
+    QgsMesh3dAveragingMethod::Method method = static_cast<QgsMesh3dAveragingMethod::Method>(
+          elem.attribute( QStringLiteral( "method" ) ).toInt() );
+    switch ( method )
+    {
+      case QgsMesh3dAveragingMethod::SingleLevelAverageMethod:
+        mAveragingMethod.reset( new QgsMeshSingleLevelAveragingMethod() );
+        mAveragingMethod->readXml( elem );
+    }
```

Comment:
It would be good to move this logic to QgsMesh3dAveragingMethod - but that can also be done in a PR later when more methods are added